File: src/main/java/com/nickcoblentz/montoya/settings/IntegerExtensionSetting.java
package com.nickcoblentz.montoya.settings;

import burp.api.montoya.MontoyaApi;
import com.nickcoblentz.montoya.MontoyaLogger;

public class IntegerExtensionSetting extends GenericExtensionSetting<Integer> {

    public IntegerExtensionSetting(MontoyaApi api, String name, String storageKey, Integer defaultValue, ExtensionSettingSaveLocation defaultSaveLocation) {
        super(api, name, storageKey, defaultValue, defaultSaveLocation);
    }

    @Override
    public void loadFromProject() {
        setCurrentValue(getApi().persistence().extensionData().getInteger(getStorageKey()));
    }

    @Override
    public void loadFromPreferences() {
        setCurrentValue(getApi().persistence().preferences().getInteger(getStorageKey()));
    }

    @Override
    public Integer parseString(String value) {
        return Integer.valueOf(value);
    }

    public void setCurrentValue(String value) {
        if(value!=null)
            setCurrentValue(parseString(value));
    }

    @Override
    public void saveToProject() {
        getApi().persistence().extensionData().setInteger(getStorageKey(),getCurrentValue().intValue());
    }

    @Override
    public void saveToPreferences() {
        getApi().persistence().preferences().setInteger(getStorageKey(),getCurrentValue().intValue());
    }
}

File: src/main/java/com/nickcoblentz/montoya/settings/IntegerExtensionSetting.kt
package com.nickcoblentz.montoya.settings

import burp.api.montoya.MontoyaApi

class IntegerExtensionSetting(
    api: MontoyaApi,
    name: String,
    storageKey: String,
    defaultValue: Int,
    defaultSaveLocation: ExtensionSettingSaveLocation
) :
    GenericExtensionSetting<Int>(api, name, storageKey, defaultValue, defaultSaveLocation) {
    override fun loadFromProject() {
        currentValue = api.persistence().extensionData().getInteger(storageKey)
    }

    override fun loadFromPreferences() {
        currentValue = api.persistence().preferences().getInteger(storageKey)
    }

    override fun parseString(value: String): Int {
        return value.toInt()
    }

    override fun setCurrentValue(value: String?) {
        if (value != null) currentValue = parseString(value)
    }

    override fun saveToProject() {
        api.persistence().extensionData().setInteger(storageKey, currentValue)
    }

    override fun saveToPreferences() {
        api.persistence().preferences().setInteger(storageKey, currentValue)
    }
}